catalog/message/schema/catalog-schema.json

```diff
@@ -1,39 +1,381 @@
 {
-	"$schema": "https://json-schema.org/draft/2019-09/schema",
-	"title": "CatalogSchema",
-	"type": "object",
-	"allOf": [
-	  {
-		"$ref": "#/definitions/Catalog"
-	  }
-	],
-	"$id": "https://w3id.org/dspace/schemas/v0.8/CatalogSchema",
-	"definitions": {
-		"Catalog": {
-			"type": "object",
-			"properties": {
-				"@context": {
-					"type": "string",
-					"const": "https://w3id.org/dspace/v0.8/context.json",
-					"required": true
-				},
-				"@type": {
-					"type": "string",
-					"const": "dcat:Catalog",
-					"required": true
-				},
-				"dcat:dataset": {
-                    "type": "array",
-                    "items": {
-    					"@type": {
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
+    "title": "CatalogSchema",
+    "type": "object",
+    "allOf": [
+        {
+            "$ref": "#/definitions/Catalog"
+        }
+    ],
+    "$id": "https://w3id.org/dspace/schemas/v0.8/CatalogSchema",
+    "definitions": {
+        "Catalog": {
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/Dataset"
+                },
+                {
+                    "$ref": "#/definitions/Resource"
+                },
+                {
+                    "properties": {
+                        "@context": {
                             "type": "string",
-                            "const": "dcat:Dataset"
+                            "const": "https://w3id.org/dspace/v0.8/context.json"
+                        },
+                        "@type": {
+                            "type": "string",
+                            "const": "dcat:Catalog"
+                        },
+                        "dcat:dataset": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/Dataset"
+                            },
+                            "minItems": 1
+                        },
+                        "dcat:record": {
```

Comment:
Why is this introduced?
> This class is optional and not all catalogs will use it. It exists for catalogs where a distinction is made between metadata about a dataset or service and metadata about the entry in the catalog about the dataset or service.

Source: [dcat v3](https://www.w3.org/TR/vocab-dcat-3/#Class:Catalog_Record)

Reply:
Same goes for `themeTaxonomy`, `hasPart`, and `homepage`. I would argue that not mentioning it here allows to still use the values. They are not mandatory anyway. It is inconsistent to name them but not the other attributes of `dcat:Catalog` (ref. superclasses [dcat:Dataset](https://www.w3.org/TR/vocab-dcat-3/#Class:Dataset) and [dcat:Resource](https://www.w3.org/TR/vocab-dcat-3/#Class:Resource)).

Reply:
> Why is this introduced?

I put it in because it's listed in the standard... If somebody uses them, it needs to be validated. We can of course decide to exclude it explicitly but as long as it's allowed, I propose to also have it in the schema.

Reply:
> I would argue that not mentioning it here allows to still use the values.

That's correct, of course. However, we should think about closing the data objects by explicitly not allowing additional attributes... Or at least document our decision.

Reply:
> it is inconsistent to name them but not the other attributes of dcat:Catalog

Did we miss any? If so, it was not intended.